### PR TITLE
Visions app

### PR DESCRIPTION
**Pull request by mike-allaway**
Developed the main 'visions' app

This PR includes code for the whole app, which has been reviewed by the research team and includes changes that they suggested during their feedback.

I'm happy to give a quick demo of it in action if needed.

### CODE REVIEW

File: django/visions/admin.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    list_display = ('__str__',
+                    'vision',
+                    'author_name',
+                    'author_email',
+                    'admin_approved',
+                    'meta_created_datetime',
+                    'meta_lastupdated_datetime')
+    list_filter = ('vision', 'admin_approved')
+    search_fields = ('response_text', 'name')
+    list_per_page = 30
```
Comment (garciaay) [inline django/visions/admin.py]: Is there a particular reason why this method has 30 entries per page, but the `VisionAdminView` has 50 entries per page?

File: django/visions/templates/visions/vision-detail.html [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+                    });
+                </script>
+            </div>
+        </div>
+        
+        <!-- List of all users' responses -->
+        <div id="response-list">
+            <h2>Responses</h2>
+
+            <div class="form-show create-form-button-link"><i class="fas fa-comments"></i> Respond to this Vision</div>
```
Comment (garciaay) [inline django/visions/templates/visions/vision-detail.html:97]: Is the space before 'Respond' intentional? I expect this to have an effect on the way the button looks.

File: django/visions/views.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        return context
+
+
+class VisionListView(ListView):
+    """
+    Class-based view for Vision list template
+    """
+    template_name = 'visions/vision-list.html'
+    queryset = models.Vision.objects.filter(admin_approved=True)
+    paginate_by = 100
```
Comment (garciaay) [inline django/visions/views.py:29]: Just double-checking the pagination, as I have seen 30, and 50 previously.

File: django/visions/views.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    """
+    Class-based view to show the vision create success template
+    """
+
+    template_name = 'visions/vision-create-success.html'
+
+
+class ResponseCreateView(CreateView):
+    """
+    Class-based view to create a new models.Response object in the database
```
Comment (garciaay) [inline django/visions/views.py]:
```suggestion
    Class-based view to create a new `models.Response` object in the database
```

File: django/visions/views.py [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+    This works by passing data to the forms.ResponseCreateForm form
```
Comment (garciaay) [inline django/visions/views.py]:
```suggestion
    This works by passing data to the `forms.ResponseCreateForm` form
```

**Review by garciaay — APPROVED** [review]
Approved!

File: django/visions/admin.py (hunk 1 above)
Comment (mike-allaway) [reply django/visions/admin.py]: I must've changed the one and not the other. Thanks, I've changed them both to 50 now

File: django/visions/templates/visions/vision-detail.html (hunk 2 above)
Comment (mike-allaway) [reply django/visions/templates/visions/vision-detail.html:97]: Yeah, the <i> tag before 'Respond' is an icon (see https://fontawesome.com/v5/search) so it's nice to separate the icon from the text a bit.

File: django/visions/views.py (hunk 3 above)
Comment (mike-allaway) [reply django/visions/views.py:29]: Yeah this is fine - the public interface can show a different amount to the admin interface

**Pull request merged**